### ISSUE TITLE
Variable interpolation don't work inside @each rules

### ISSUE THREAD

**Issue by ghost** [sass/sass#132]
$ck: #66cc00;
$es: #3366cc;
$fr: #ff6c00;
$bd: #cc0000;
$wt: #FFDB00;

@each $brand in ck, es, fr, bd, wt{
  .#{$brand}_bg {
    background-color: $#{$brand};  // <-- I want to call $ck, $es, $fr, $bd, $wt
  }
}

**Comment by ChristianPeters** [comment]
+1 

Variable interpolation does not work at all. 
A common workaround is a lookup function (see links below).

I started a discussion about this issue a while ago: [Interpolation of Variable Names - Sass Google Group](http://groups.google.com/group/sass-lang/browse_thread/thread/ba9afbf1ccab5b52/3562720f3c4b9cff)

And I came across this problems multiple times since then.

If you use Rails 3.1 with asset pipeline, you could use ERB as a preprocessor of Sass, as stated in [this discussion about Sass variable interpolation at Stackoverflow](http://stackoverflow.com/questions/6010013/is-it-possible-to-reference-a-variable-with-an-interpolated-string).

**Comment by jonjaques** [comment]
You can use `@for` to loop through lists.

``` scss
$brands: a, b, c;
$val: blue, red, yellow;
@for $i from 1 through length($brands) {
  .#{nth($brands, $i}-color {
    color: nth($val, $i);
  }
}
```

would give you

``` css
.a-color { color: blue }
.b-color { color: red }
.c-color { color: yellow }
```

Otherwise a more clever but hackish way would be to create a lookup function.

``` scss
$brands: a, b, c;
$brand_colors: blue, red, yellow;

@function getValue($prop, $val, $search) {
    $n1: index($prop, $search);
    $n2: index($val, $search);
    @if($n1) {
        @return nth($val, $n1);
    } @else {
        @return nth($prop, $n2);
    }
}

@each $brand in $brands {
    .#{$brand}-color {
        color: getValue($brands, $brand_colors, $brand);
    }
}

.test-twoway-lookup {
    color: getValue($brands, $brand_colors, red);
}

```

Which would yield:

``` css
.a-color { color: blue; }
.b-color { color: red; }
.c-color { color: yellow; }
.test-twoway-lookup { color: en; }
```

Anyways, the key is you have to keep the lists as ordered pairs - so not quite as good as a hash.

**Comment by chriseppstein** [comment]
I think all the cases I've seen for this feature are better handled by adding an associative array (hash) data structure.

**Comment by jonjaques** [comment]
YES! See issue #315 (closed). My pseudo code is a little bit off the mark, but it's mostly because of limitations I'm facing on my end. The question is what the hash can hold. Would there be a strict CSS property to value relationship, or could you store anything? Also, how would mixins (thinking of Compass CSS3 specifically) work? 

``` scss
$myHash: [
   color => $action-color,
   background => $action-background,
   border-radius => $action-radius
   // @include border-radius($action-radius)   This wouldn't work/make sense
]
```

**Comment by jstoller** [comment]
Here is my use case, [copied from this issue](https://github.com/nex3/sass/issues/329). I'm to extend the zen-grid-item mixin so it easily supports multiple grids, based on two independent variables: $grid-name and $grid-size.

```
$default-grid-name: standard;
$default-grid-size: full;

$standard-full-grid-column-count: 10;
$standard-full-grid-gutter-width: 20px;
$standard-full-grid-width: 980px;

$standard-small-grid-column-count: 2;
$standard-small-grid-gutter-width: 20px;
$standard-small-grid-width: 100%;

$front-full-grid-column-count: 12;
$front-full-grid-gutter-width: 20px;
$front-full-grid-width: 980px;

$front-small-grid-column-count: 2;
$front-small-grid-gutter-width: 20px;
$front-small-grid-width: 100%;

@mixin my-grid-item (
  $grid-name: $default-grid-name,
  $grid-size: $default-grid-size,
  $column-span: $#{$grid-name}-#{$grid-size}-grid-column-count, // <- note variables
  $column-position: 1,
  $float-direction: $zen-float-direction,
  $column-count: $#{$grid-name}-#{$grid-size}-column-count,    // <- note variables
  $gutter-width: $#{$grid-name}-#{$grid-size}-gutter-width,    // <- note variables
  $grid-width: $#{$grid-name}-#{$grid-size}-grid-width,        // <- note variables
  $box-sizing: $zen-box-sizing,
  $reverse-all-floats: $zen-reverse-all-floats,
  $auto-include-item-base: $zen-auto-include-item-base
) {
  @mixin zen-grid-item (
    $grid-name,
    $grid-size,
    $column-span,
    $column-position,
    $float-direction,
    $column-count,
    $gutter-width,
    $grid-width,
    $box-sizing,
    $reverse-all-floats,
    $auto-include-item-base
  )
}
```

This woud allow me to something like...

```
#content {
  @include my-grid-item(standard, full, 6, 3);
}
.front-page #content {
  @include my-grid-item(front, full, 8, 3);
}
@media screen and (max-width : 480px) {
  #content {
    @include my-grid-item(standard, small);
  }
  .front-page #content {
    @include my-grid-item(front, small);
  }
}
```

This example shows only two grid names and two grid sizes, but in practice there will be several more name/size combinations. And there is no guarantee that every grid name will have the same sizes associated with it. Accomplishing this without variable interpolation within variable names seems ridiculously complex.

**Comment by chriseppstein** [comment]
I think a map-based data structure would be more straightforward than dynamic variables.

**Comment by jstoller** [comment]
A map-based data structure seems WAY more complicated when you're dealing with multiple, independent variables. I don't want to be setting up complex multi-dimensional arrays in my Sass. My above example makes it easy. I just set my variables once, and then I reference them. Adding more variables is easy. No arrays involved. And I can list the variables in whatever order makes sense to people, rather than an array, which would only make sense to a computer, thus increasing usability.

**Comment by chriseppstein** [comment]
I didn't say an array.

**Comment by chriseppstein** [comment]
I'm thinking something like:

``` scss
$front-small-grid: {
  name: small,
  position: front,
  count: 2,
  gutter: 20px,
  width: 100%
};

@mixin my-grid-item (
  $grid: $default-grid,
  $column-position: 1,
  $float-direction: $zen-float-direction,
  $box-sizing: $zen-box-sizing,
  $reverse-all-floats: $zen-reverse-all-floats,
  $auto-include-item-base: $zen-auto-include-item-base
) {
  @include zen-grid-item (
    fetch($grid, name),
    fetch($grid, size),
    $column-span,
    $column-position,
    $float-direction,
    fetch($grid, count),
    fetch($grid, gutter),
    fetch($grid, width),
    $box-sizing,
    $reverse-all-floats,
    $auto-include-item-base
  )
}

@include my-grid-item($front-small-grid);
```

**Comment by danielbayerlein** [comment]
**+1** for this feature.

**Comment by kuzin** [comment]
+1 This could be very useful in keeping things DRY

**Comment by timmfin** [comment]
+1

**Comment by bmatzner** [comment]
+1

**Comment by danielbayerlein** [comment]
Good news in #463!

**Comment by chriseppstein** [comment]
closing. We're going to add maps.

**Comment by noAlvaro** [comment]
Sorry, I just cannot realize how this map/hash approach would work in some examples.

Consider a simple @mixin to deal with compass sprite creation:

```
@mixin declare-sprite($folder, $layout: vertical)
    $#{$folder}-layout: $layout
    @import "#{$folder}/*.png"
```

How could hash approach be better than variable interpolation?

**Comment by chriseppstein** [comment]
In this case, you should simply call the sprite function directly and then you don't need all the magic variables.

```
sprite("#{$folder}/*.png", $layout: $layout)
```

**Comment by noAlvaro** [comment]
Ok then.

**Comment by noAlvaro** [comment]
Well, sorry maybe I'm a lazy head... so could we try this again?
Changing examples, let's imagine a `sass` component which theme should be served by different sprites, according to a given ID.
Something like that:

```
@each $theme in aero, space, nature
    @import "themes/#{$theme}-images/*.png"
    .component##{$theme}
        @include #{$theme}-images-sprite(skin-texture)
```

How could we make it with hashes?

**Comment by Snugug** [comment]
Variable interpolation absolutely works inside @each loops, but variable interpolation never works in @import statements or mixin calls.

**Comment by noAlvaro** [comment]
Yeah, ok. The code wouldn't work, indeed. But my issue here is about testing capabilities of different approaches against each other. Just couldn't figure how to `hash`/`map` it out yet.

**Comment by jstoller** [comment]
@chriseppstein, is there an issue where the map feature you mention is being discussed, and roughly when do you think we'll see it in a release?

**Comment by michaek** [comment]
I'm also anxious to see Hash come to SASS. The lack of variable interpolation, or really any way to get a stored value based on an arbitrary string, has definitely held me back a number of times.

**Comment by noAlvaro** [comment]
With Rails, I've been using `.sass.erb` extension to hardcode sass via `<%= ruby_code %>`, using Sprockets.
I found this is actually better than make use of variable interpolations, because sometimes is necessary to pull data/variable names from the database...

**Comment by michaek** [comment]
I'm not using Rails in this case, and I'm certain that an explicit value from somewhere else in the stack isn't always the right answer. :)

**Comment by chriseppstein** [comment]
Maps are slated for Sass 3.3 which will be coming out soon.

**Comment by michaek** [comment]
That's awesome news, @chriseppstein. Thanks!

**Comment by johnnyshields** [comment]
Maps are great, but I think there's still a use case for variable interpolation. For example, I want to do:

``` sass
$statuses: "cancelled", "rejected", "pending", "confirmed", "done";

$color-status-cancelled: #e74c3c;
$color-status-rejected:  #f39c13;
$color-status-pending:   #b04f00;
$color-status-confirmed: #870000;
$color-status-done:      #48F;

@each $status in $statuses {
  .#{status} > div { background-color: $color-status-#{$status}; }
}
```

**Comment by Arcovion** [comment]
@johnnyshields why not use this?

``` sass
$statuses: (cancelled: #e74c3c, rejected: #f39c13, pending: #b04f00, confirmed: #870000, done: #48F)

@each $status, $color in $statuses 
  .#{$status} > div 
    background-color: $color 
```

Then use `map-get()` instead of variables:

``` sass
.other > div
  background-color: map-get($statuses, cancelled)
```

**Comment by johnnyshields** [comment]
@Arcovion cool! That will get the job done for now I guess.

If I had more variables like `$color-#{$status}-background`, `$color-#{$status}-text`, etc variable interpolation would still be nice (I could do arrays or nested hashes as the hash values I suppose...)

**Comment by hpohlmeyer** [comment]
Any chance to still see variable name interpolation some time in the future?

In general most of the use cases for it can be solved with maps, but there are some cases which would benefit from it.
In my case this would be the usage of the SUIT naming convention. I am working with flat named variables in the global space and would like to use them in mixing like this:

``` scss
$--bpMobileMaxWidth: 320px;

@include breakpoint('Mobile') {
  body { max-width: $--bpMobileMaxWidth; }
}
```

There is no way to search for the variables from inside the mixin without interpolating the variable name like this:

``` scss
@mixin breakpoint($name: null) {
  @if $--bp#{$name}MaxWidth { … }
}
```

I created a gist with a full code example and a map solution to illustrate this case:
https://gist.github.com/hpohlmeyer/fde8742dfc8c07f6a463

I know, that in most cases the maps solution would be better, but it seems to me, that this is an often requested feature for special cases. Is there any specific reason not to include it? (especially when this is the only situation `#{}` does not work as expected) I think a lot of people would appreciate it.

**Comment by chriseppstein** [comment]
> I think a lot of people would appreciate it.

I think many more would curse it because it would very frequently misused. Maps are the way to do this.

**Comment by hpohlmeyer** [comment]
Hm, no implementation because of high potential for misuse?
I can understand it, but it makes it hard to follow variable naming conventions …

**Comment by chriseppstein** [comment]
I would question those conventions then.

**Comment by DrSammyD** [comment]
How about allowing map-get on the global scope object?

**Comment by miraage** [comment]
:up:

**Comment by klangwiedergabe** [comment]
@chriseppstein what about the cases when you need this feature? Are we all have to be screwed because other people may misuse it? That's bad argument for anything. Nothing should be done in the world if we always asume people will misuse something.

Maybe we could try to see the bright side here, how many people would use it correctly and benefit from it? :dancers:

**Comment by cimmanon** [comment]
@klangwiedergabe There is no case where you need this "feature".  In fact, it is a gross misfeature of PHP, and it was copied from Perl.

Do you know what happens when you do searches like "how do you use variable variables in language X" (if it was a question asked on SO, it will usually be phrased to include "like I can with PHP") where language is anything other than PHP or Perl?  You see disparaging comments about PHP/Perl and that you're supposed to use mappings instead.

**Comment by klangwiedergabe** [comment]
But I inject a plugin in a project, and it give me variables already named by.. the plugin. So if I want to loop those variables I need a feature like interpolation inside variables. I don't wan't to write my own plugin to make right nor people saying me that I shouldn't use any and write all by myself. I actually can't do it, because the plug do things that I don't even know how to do, and then it gives me the variables. It's awesome but it would be awesome to have "this feature" too.

Sass was created to speed things up not to late them down, right?

**Comment by cimmanon** [comment]
If you were meant to iterate over the variables, they would be contained in a structure that can be iterated over (mapping, list).  The problem seems to be with the library you're using, not Sass.  You should be petitioning with them to make changes you desire.

**Comment by johnnyshields** [comment]
What if we implemented this as a Sass function, something like this:

```
get-var("$color-status-%{1}", $status)
```

(This is inspired by Python syntax `getattr(obj, "name")(args)`)

**Comment by davidkpiano** [comment]
@johnnyshields Use a map:

``` scss
$color-status: (...); // a map of color statuses

$foo: map-get($color-status, $status);
```

It's even fewer characters. Dynamic variables means that the variables are grouped/related somehow, which means using a list/map makes much more sense.

**Comment by nazartokar** [comment]
That would be a great feature.

**Comment by carolirod** [comment]
@chriseppstein It would be great if it's implemented in case there is an already BIG and existent project which doesn't have the set of variables in a map and needs to use the already existing variables' names (because otherwise would need to change throughout all the project the way of calling the map... which it can be done but... you know), just saying.

**Comment by ArmorDarks** [comment]
@chriseppstein

> I would question those conventions then.

Forcing convention on user is quite bad things, isn't it? Didn't you think that after 1 year you might find your own convention, which your product forced on user, to be outdated, irrelevant?

---

Regarding issue in question, I'd say that current implementation of Sass sometimes just forces you to use variables in places where would be more logical to use structured objects, like Sass map.

The reason is quite simple — very frustrating way to assign and extract values.

Just compare Sass way with JavaScript's (coffee):

``` coffee
gridSettings = {
  maxWidth: 1200px
  minWidth: 200px
}

gridSettings.gutter  = 2px
gridSettings.columns = 12

# ...
# Easy way to set or expand data
```

And with JavaScript-like usage of that data would be much easier:

``` Sass
mixin grid(
  $gutter  : $gridSettings.gutter,
  $columns : $gridSettings.columns 
)
```

In the end, you just don't feel big difference while working with objects or plain variables in JavaScript.

So, what we already have in Sass, compare readability against JavaScript example:

``` sass
$gridSettings = (
  maxWidth: 1200px,
  minWidth: 200px
)

# now we need to set our new properties... but wait!
# Sass has no native way for set, only merge!
# ok... with functions... now lookup... *okayface*

map-merge($gridSettings, (gutter : 2px));
map-merge($gridSettings, (columns : 12));
```

and now let's pass it to our mixin:

``` sass
mixin grid(
  $gutter  : map-get($gridSettings, 'gutter'),
  $columns : map-get($gridSettings, 'columns')
)
```

huh... that was really painful, isn't it? Now, imagine when you need to have few sets of grid, which means 2-lvl map at best. Working with it will be even harder, even with adopted for that purpose custom function — you will spend tons of time just on typing for doing simple things, and while it can be avoided with snippets, code readability will be average at best.

**Comment by davidkpiano** [comment]
@ArmorDarks Playing devil's advocate here - but I really wouldn't compare Sass maps to plain JavaScript objects. Rather, they're more comparable to [ES6 Maps](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map), in which there is not much difference:

_ES6 Map_

``` js
let gridSettings = new Map();
gridSettings.set('maxWidth', 1200);
gridSettings.set('minWidth', 200);
gridSettings.set(['a', 'b', 'c'], 'foo');
gridSettings.set({bar: baz}, 'qux');

let minWidth = gridSettings.get('minWidth');
```

_Sass Map_

``` scss
$grid-settings: (
  'maxWidth': 1200px,
  'minWidth': 200px,
  ('a', 'b', 'c'): 'foo',
  ('bar': 'baz'): 'qux'
);

$min-width: map-get($grid-settings, 'minWidth');
```

The SCSS syntax can even be _more_ terse than the JS syntax.

**Comment by cimmanon** [comment]
@ArmorDarks I don't see what any of that has to do with allowing/disallowing interpolation of variable names.  If you want to make a point about how much you dislike the syntax for modifying/accessing elements of a mapping, you're in the wrong thread (see: https://github.com/sass/sass/issues/1349).

**Comment by ArmorDarks** [comment]
@davidkpiano Thanks for your reply.

In comparison with JavaScript maps — yes, Sass doing pretty same thing. But we shouldn't forget why JS maps using new `get` and `set` methods for maps instead of typical JS lookup like `gridSettings.minWidth` — it's because JS's map key can be _anything_, even another object.

That is something you probably won't do in Sass very often (though, to be honest, I didn't know you even can hold objects as keys in Sass maps), so plain lookup would be better, faster.

@cimmanon Sorry for the confusion. I didn't mean to start offtopic. I tried to explain why some developers keep asking for variables interpolation — because current way of working with maps isn't comfortable. It is something that other devs just were too shameful to say, I guess, because it won't sound as reasonable argument.

I completely agree with maintainers that variable interpolation isn't needed. If you're iterating over something, let's say, settings, generating something new based on them — you should use map (object, dict — depending on you language).

But with Sass current ways it just doesn't work well. It's ok when you're writing Sass framework. In fact, usually nobody should dive too much into it's internals. But mess starts when you need to use over and over and over framework's values on external levels. You just can't do that quickly.

All this doesn't mean, though, that variables interpolation should be implemented.

**Comment by whitetrefoil** [comment]
Just a little personal opinion, the reason I'm desiring dynamic variable name is this: #216.

The functionality of Map maybe indeed enough to replace dynamic variable name, but just too difficult to read in plain SASS.

``` sass
// I'm looking for something like this for long:

$blue-100: #xxx
$blue-200: #yyy
// ...
$blue-1000: #zzz
$red-100: #aaa
$red-200: #bbb
// ...

@function color($hue, $level)
  @return $#{$hue}-#{$level}
```

I'm using Map to get this, but it needs a terribly long line to define...  And it's almost impossible to read

``` sass
$color: (blue: (100: #xxx, 200: #yyy, ...), red: (...), yellow: (...), green: (...), cyan: (...))
```

**Comment by rosskevin** [comment]
Yet another case for this.  I'm not in control of the upstream library (bootstrap), and I would like to override one property in a dry way.

``` scss
  .btn-primary {
    color: $btn-primary-color;
  }
  .btn-secondary {
    color: $btn-secondary-color;
  }
  .btn-info {
    color: $btn-info-color;
  }
  .btn-success {
    color: $btn-success-color;
  }
  .btn-warning {
    color: $btn-warning-color;
  }
  .btn-danger {
    color: $btn-danger-color;
  }
```

This could easily be written as a loop over a state list with dynamic interpretation of variables.  
1.  I don't see a problem with bootstrap's naming, actually, it is very consistent and makes much sense (the perfect case for dynamic retrieval of variables).
2. To use maps, I would have to add all of bootstrap's configurable variables that I need to a map, then look them up.  That is a lot of keys/values to maintain.  This is net more code, not less, and not easier to read.
3.  This is **not the only case** I need to override these states but with a _different_ variable, so my code is either made up of these big simple but repetitive blocks, or I create a gratuitous number of variable maps.

While the argument that "this could be misused" may be true, _anything_ can be misused and denying good uses to try and keep out the bad simply punishes those of us that know how to write good code.

**Comment by joelcoxokc** [comment]
Has this been implemented Yet. Ive been following this chain for a while. waiting for this feature. 

Regardless of the use cases, the should be included. 
When building Libraries and tools for other developers who don't understand sass, It makes it difficult to teach them how to use a Library with `map-get` and then how to create a function that will traverse nested keys. 

I love maps and a I use them every day. 

But it would be super nice to be able to provide global interpolating variables

like `sidebar-#{$screen}-#{$platform}-width`;

instead of having a map that I have to require the consumers to iterate through.

**Comment by jstoller** [comment]
Agreed. This would be a big UX improvement.

**Comment by chriseppstein** [comment]
> While the argument that "this could be misused" may be true, anything can be misused and denying good uses to try and keep out the bad simply punishes those of us that know how to write good code.

Sure. But I think we can also agree that for any given feature there is an _amount of benefit_ and an _amount of cost_ that will be born not just by the original author but by subsequent author and editors of the code down the road. It is our _opinion_ that the amount of benefit that you get by drying up code with this feature is not justified by the amount of confusion it generates downstream. I'm sure you disagree, and we're ok with that. We're ok with making some percentage of great use cases harder but **keeping the overall sense in the general public that Sass files are MORE maintainable than CSS**. It's not a science; we can't measure these things in an absolute sense. But I've been programming for almost 30 years now, and my experience tells me that variable interpolation is a net negative feature.

**Comment by ArmorDarks** [comment]
> keeping the overall sense in the general public that Sass files are MORE maintainable than CSS

Why I'm getting feeling that during last year Sass team overthinking this aspect of Sass way too much...

**Comment by chriseppstein** [comment]
> Why I'm getting feeling that during last year Sass team overthinking this aspect of Sass way too much...

Probably because you are not seeing all the feedback we see from across the community.

**Comment by rachelreveley** [comment]
I'm just moving over to  Sass from Less, which is harder than some sites claim, it's not just convert @ to $. 

What would I do if I wanted to create a loop that generates a numbered list of variables that incrementally changed a colour programmatically?

Basically I have been trying to use the code here http://codepen.io/thebeckyhamm/pen/szEyK (The Mixing one at the bottom) to create seven steps between 2 specified colours and output those as variables rather than as CSS rules.

**Comment by ArmorDarks** [comment]
@rachelreveley you definitely should use Sass maps or Sass list for this. Sass list seems to me to be best candidate for that role.

You can populate stored in variable (let's say `$my-list`) list with 7 values during `@for` loop, and then access those values with `nth($my-list, 1)`, `nth($my-list, 2)`, `nth($my-list, 3)`, `nth($my-list, 4)`, `nth($my-list, 5)`, `nth($my-list, 6)`, `nth($my-list, 7)`.

Here is working example:

``` scss
$darken-color: #0046C5;

$steps-list: ();

@for $i from 1 through 10  {
  $steps-list: append($steps-list, darken($darken-color, ($i * 2%)))
}

// This will make `$step-list` return following list:
// #0042bb
// #003fb1
// #003ba6
// #00389c
// #003492
// #003088
// #002d7e
// #002973
// #002569
// #00225f

// use `nth($steps-list, {{index of value}})` to access colors

.bg1 {
  background-color: nth($steps-list, 1)
}

.bg2 {
  background-color: nth($steps-list, 2)
}

.bg3 {
  background-color: nth($steps-list, 3)
}

// ...
```

**Comment by cimmanon** [comment]
@rachelreveley Functions are a cleaner way to handle this situation:

``` sass
$my-color: #0046C5;

@function color-step($step, $color: $my-color) {
    @return darken($color, $step * 2%);
}

.foo {
    color: color-step(1);
}

.bar {
    color: color-step(5);
}
```

**Comment by rachelreveley** [comment]
Thanks @cimmanon and @ArmorDarks 

That does make more sense to me at this point. I have modified the function to do a mix between 2 colours http://codepen.io/rachelreveley/pen/zqyBRO

**Comment by joelcoxokc** [comment]
@rachel

@for $i from 1 through 12 {

$i === the incrementing number.

color: create-color($i)

}

**Comment by seeliang** [comment]
Sorry, my understanding so far is we can't do this 

```
@mixin font-load ($name, $style: none) {

  @if $style == none {
    %font-#{$name} {
      font-family: $font-#{$name};
    }
  } 
}

```

and we need a workaround.

Right?

**Comment by moarsel** [comment]
I ran into this issue, but recently found a workaround:

Initially I had:

```
@for $i from 1 through $grid-columns {
      .col--#{$breakpoint-name}-#{$i} { ... }
}
```

Then changed it so it was interpolated first in a variable:

```
@for $i from 1 through $grid-columns {
      $col-class-name: ".col--#{$breakpoint-name}-#{$i}";
      #{$col-class-name} { ... }
}
```

Hope this helps someone.

**Comment by telinc1** [comment]
I've got no idea why the developers are so insistent on not implementing this, it's just turning into pure laziness at this point. It would be a good feature and I already have another use case for it:

```
@mixin flex-break($breakpoint) {
    &.flex-break-#{$breakpoint} {
        @include flex(initial);
        width: 100%;
        display: none;

        @media(max-width: ??){
            display: block;
        }
    }
}
```

As you can see, in the above mixin I need the breakpoint's name to add it to the class name, however I also need the variable for it in the media query part of the mixin. In other parts of my code, I do directly use the `$breakpoint-*` variables. Does all of this **USE MAPS** stuff mean I have to replace all of my media queries? What looks better, `$breakpoint-logo` or `map-get($breakpoints, "logo")`? So it's either make your code look ugly or define everything twice. Thanks, Sass, you really do make my workflow better.

**Comment by ArmorDarks** [comment]
@telinc1 yes, from every perspective it would be wiser to use maps in this case. About ugly syntax for accessing maps value — it's more related to this issue https://github.com/sass/sass/issues/1349, though, unfortunately, we'll never see it implemented.

For now I'd recommend to make dedicated function, which will return value of stated breakpoint from predefined in that function map. `breakpoint('logo')` will make code cleaner.

**Comment by 7oque** [comment]
Well, it would have been nice if I could do this;

```
@for $n from 1 through 8 {
  .item-#{$n} {
    @include sprite( $item-#{n} );
  }
}
```

But this would be bad practice?

**Comment by ArmorDarks** [comment]
@Loque- Just keep your sprites in map and it will make your life much easier :) And you won't need to iterate based on enigmatic `1 through 8`, you could use your sprites map to iterate exactly based on it.

**Comment by 7oque** [comment]
Thank you for the reply @ArmorDarks - if I wasn't trying to plugin an existing sprite tool (or had the time to edit it/make my own), then yes! (using https://github.com/sprity/sprity-sass).

Unless you know of a good CLI node sprite generator that outputs sass maps instead?

I still don't understand what is so bad about my example above though - surely it would be really nice and useful to be able to do that?

**Comment by ArmorDarks** [comment]
@Loque- You can try add your own template, which will generate not variables, but map with sprites (https://github.com/sprity/sprity-sass/issues/5).  And here is another example: https://github.com/sprity/sprity-sass/issues/2.

I can't recommend anything other, since we never needed spriting utility for pure node. We've been always using [grunt-spritesmith](https://github.com/Ensighten/grunt-spritesmith) and it's capable of generating maps. You can try adopt [it's engine](https://github.com/Ensighten/spritesmith).

**Comment by 7oque** [comment]
Thank you @ArmorDarks, I was just looking at the templates also. For the moment, I'll simply duplicate each line as it is not much hassle (I am fixing someone else's build by implementing spriting). 

Spritesmith has a CLI which I could probably use (I am currently pulling out grunt and gulp in favour of npm scripts), thanks so much for the suggestions, appreciated. If I hit the issue again I'll definitely pick up either of these solutions.

Not that this probably means anything, but I love a bit of var interpolation inside of loops : ) - maybe I am just javascript born and bred.

**Comment by ArmorDarks** [comment]
@Loque-

> Not that this probably means anything, but I love a bit of var interpolation inside of loops : ) - maybe I am just javascript born and bred.

Well, JavaScript doesn't have interpolation for variables names...

**Comment by 7oque** [comment]
Hah, true, no idea what I was thinking there ~

**Comment by kazelsama** [comment]
<pre>@each $var in (a,b,c) {
    .#{$var}{
        background:$$var;
    }
}</pre>

no? :)

**Comment by JakedUp** [comment]
This would be so much more convenient than maps! Since I'm using multiple libraries, such as Bootstrap, which does not store it's variables in maps, I have no simple way to access those variables in a shorthand fashion.

```
@function color($variable) {
  @if (global-variable-exists("brand-"+$variable)) {
    @return #{"brand-"+$variable};
  } @else {
    @return null;
  }
}

$brand-primary: #000;
$brand-secondary: #555;

a {
  color: color("primary"); // #000

  &:hover {
    color: color("secondary"); // #555
  }
}
```

Virtually every other language out there does variable interpolation, including JavaScript:

```
function color(color){
  return window["brand_"+color];
}

brand_primary = "#000";
brand_secondary = "#555";

color("primary"); // "#000"
color("secondary"); // "#555"
```

**Comment by moarsel** [comment]
Try creating a variable first. This works in most of the cases in this thread:

```
$colorname: "brand-"+$variable;

global-variable-exists($colorname);
...
```

**Comment by JakedUp** [comment]
@moarsel, it's not working. $colorname equals the name of the variable, not the value of the variable. Need the value.

**Comment by luxueyan** [comment]
I think it is a good function，why not support it！ It have been a long time for resolved this issue

**Comment by dejongnj** [comment]
I'm running into a similar issue in trying to interpolate variable names. I'm importing css which has a lot of constants anyway which have a structure:
$brandName-attribute-size
e.g.
$brandName-font-size-s, $brandName-font-size-m, $brandName-font-size-l etc.

If I want to create classes based on the different font sizes set to those particular constants, it would be trivially easy to do based on variable interpolation.  Using maps seems like it would result in a lot of redundant work.

**Comment by ange007** [comment]
A very necessary function. 
Even in LESS, she has been there for a long time, but there is still no :(

**Comment by joelcoxokc** [comment]
I don’t think making a comparison between Sass and Less is necessary, contructive, or even logically accurate.

**Comment by ange007** [comment]
> I don’t think making a comparison between Sass and Less is necessary, contructive, or even logically accurate.

There are not many CSS preprocessors, so I only compare with what I know about.
And just starting from this, the tool for the task is chosen.

**Comment by johnnyshields** [comment]
> I don’t think making a comparison between Sass and Less is necessary, contructive, or even logically accurate.

Sure it is. LESS and SASS do fundamentally the same job. Why not be open to hearing about good ideas that evolved in the LESS world?

**Comment by joelcoxokc** [comment]
I apologize I didn’t mean to offend anybody. I will delete my message. 
Less is great. 
Less also makes it impossible to iterate... so this really isn’t that big of an issue in Less. 

Sass needs variable interpolation. 
They made it clear that it is not in on their road map. 
Sass still need variable interpolation.

**Comment by nex3** [comment]
The discussion here isn't going anywhere productive, so I'm going to lock this issue. Maps are the recommended way to dynamically access values by name. If anyone's having trouble expressing their use-case using maps, I recommend [asking for assistance on StackOverflow](https://stackoverflow.com/questions/ask?tags=sass).
